src/auth/screens/login.screen.js

```diff
@@ -229,130 +243,120 @@ class Login extends Component {
   }
 
   render() {
-    const { locale, isLoggingIn, isAuthenticated } = this.props;
+    const { locale, isLoggingIn } = this.props;
 
     return (
       <ViewContainer barColor="light">
-        {!isAuthenticated &&
-          !isLoggingIn &&
-          this.state.asyncStorageChecked && (
-            <View style={styles.container}>
-              <Modal
-                animationType="slide"
-                onRequestClose={() => null}
-                visible={this.state.modalVisible}
-                style={styles.container}
-              >
-                <View style={styles.modalContainer}>
-                  <View style={styles.browserSection}>
-                    <WebView
-                      source={{
-                        uri: `https://github.com/login/oauth/authorize?response_type=token&client_id=${CLIENT_ID}&redirect_uri=gitpoint://welcome&scope=user%20repo&state=${stateRandom}`,
-                      }}
-                      onLoadStart={e => this.toggleCancelButton(e, true)}
-                      onLoadEnd={e => this.toggleCancelButton(e, false)}
-                      onNavigationStateChange={e =>
-                        this.onNavigationStateChange(e)}
-                      renderLoading={() => this.renderLoading()}
-                      startInLoadingState
-                    />
-                  </View>
-                  <View style={styles.miniSection}>
-                    <Button
-                      title={translate('auth.login.cancel', locale)}
-                      buttonStyle={styles.button}
-                      disabled={this.state.cancelDisabled}
-                      textStyle={styles.buttonText}
-                      onPress={() =>
-                        this.setModalVisible(!this.state.modalVisible)}
-                    />
-                  </View>
-                </View>
-              </Modal>
-              <View style={styles.miniSection}>
-                <Image
-                  style={styles.logo}
-                  source={require('../../assets/logo.png')}
+        {this.shouldDisplayIntro() && (
```

Comment:
How about moving the code to a separate component, such as `Intro`?

```js
{this.shouldDisplayIntro() && <Intro />}
```

Reply:
I'm not sure about this, it would be used only for the login screen (+comment below)